Shape_detection/include/CGAL/Polygon_mesh_processing/region_growing.h

```diff
@@ -0,0 +1,338 @@
+                      must be available in `PolygonMesh`.}
+    \cgalParamNEnd
+    \cgalParamNBegin{geom_traits}
+      \cgalParamDescription{an instance of a geometric traits class}
+      \cgalParamType{a class model of `Kernel`}
+      \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+      \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+    \cgalParamNEnd
+    \cgalParamNBegin{maximum_distance}
+      \cgalParamDescription{the maximum distance from a point to a plane}
+      \cgalParamType{`GeomTraits::FT`}
+      \cgalParamDefault{1}
+    \cgalParamNEnd
+    \cgalParamNBegin{maximum_angle}
+      \cgalParamDescription{the maximum angle in degrees between
+      the normal of the supporting planes of adjacent faces}
+      \cgalParamType{`GeomTraits::FT`}
+      \cgalParamDefault{25 degrees}
+    \cgalParamNEnd
+    \cgalParamNBegin{cosine_value}
+      \cgalParamDescription{the cos value computed as `cos(maximum_angle * PI / 180)`,
+      this parameter can be used instead of the `maximum_angle`}
+      \cgalParamType{`GeomTraits::FT`}
+      \cgalParamDefault{`cos(25 * PI / 180)`}
+    \cgalParamNEnd
+    \cgalParamNBegin{minimum_region_size}
+      \cgalParamDescription{the minimum number of points a region must have}
+      \cgalParamType{`std::size_t`}
+      \cgalParamDefault{1}
+    \cgalParamNEnd
+  \cgalNamedParamsEnd
+ */
+template<
+typename PolygonMesh,
+typename RegionMap,
+typename CGAL_NP_TEMPLATE_PARAMETERS>
+std::size_t
+region_growing_of_planes_on_faces(
+  const PolygonMesh& mesh,
+  RegionMap region_map,
+  const CGAL_NP_CLASS& np = parameters::default_values())
+{
+  typedef typename GetVertexPointMap < PolygonMesh, CGAL_NP_CLASS>::const_type VPM;
+  typedef typename GetGeomTraits<PolygonMesh, CGAL_NP_CLASS>::type  Traits;
+
+  typedef boost::graph_traits<PolygonMesh> Graph_traits;
+  typedef typename Graph_traits::face_descriptor face_descriptor;
+
+  using parameters::choose_parameter;
+  using parameters::get_parameter;
+  namespace RG_PM = CGAL::Shape_detection::Polygon_mesh;
+
+  using Neighbor_query = RG_PM::One_ring_neighbor_query<PolygonMesh>;
+  using Region_type = RG_PM::Least_squares_plane_fit_region<Traits, PolygonMesh, VPM>;
+  using Sorting = RG_PM::Least_squares_plane_fit_sorting<Traits, PolygonMesh, Neighbor_query, VPM>;
+  using Region_growing = CGAL::Shape_detection::Region_growing<Neighbor_query, Region_type>;
+
+  Neighbor_query neighbor_query(mesh);
+  Region_type region_type(mesh, np);
+  Sorting sorting(mesh, neighbor_query, np);
+  sorting.sort();
+
+  std::vector<typename Region_growing::Primitive_and_region> regions;
+  Region_growing region_growing(
+    faces(mesh), sorting.ordered(), neighbor_query, region_type);
+  region_growing.detect(CGAL::Emptyset_iterator());
+
+  for (face_descriptor f : faces(mesh))
+    put(region_map, f, get(region_growing.region_map(), f));
```

Comment:
Seems like it should be possible to use `region_map` to be used  directly as the pmap of `region_growing`?